### PR TITLE
TestConductorHandle to remove test boilerplate

### PR DESCRIPTION
**Pull request by maackle**
Introduces a `test_conductor` module with some cool stuff:

- `TestConductorHandle` wraps a normal `ConductorHandle` to provide:
    - An opinionated app installer which takes the cross product of a handful of agents and a handful of DNAs, to produce the necessary Cells
        - in retrospect maybe this could have just worked on one agent at a time, and hence one app at a time
        - but, it is opinionated, and if we need a different opinion, we can write a different app installer method on the `TestConductorHandle`
    - A way to call zomes with minimal boilerplate, handling the ExternInput/Output and SerializedBytes dance.
- `TestCell`, a partial application of a call to a Cell
- `TestZome`, a partial application of a call to a Zome in a TestCell

Also included:

- Some helper methods to construct a DnaFile directly from a collection of Zomes
- A `TestAgents` construct which generates AgentPubKeys in a stream, a Vec, or tuples from arity 1-3.

I also unleashed these tools on the initial InlineZome integration test, and also chose another big test at random to refactor (`ribosome_authorized_call`)

### CODE REVIEW

File: crates/holochain/src/core/ribosome/host_fn/capability_grants.rs [hunk 1]
```diff
@@ -172,272 +136,94 @@ pub mod wasm_test {
...
-        match output {
-            ZomeCallResponse::Ok(guest_output) => {
-                let response: SerializedBytes = guest_output.into_inner();
-                let inner_response: ZomeCallResponse = response.try_into().unwrap();
-                // the inner response should be unauthorized
-                assert_matches!(inner_response, ZomeCallResponse::Unauthorized(_, _, _, _));
-            }
-            _ => unreachable!(),
-        }
-
```
Comment (maackle) [inline crates/holochain/src/core/ribosome/host_fn/capability_grants.rs:336]: Behold: boilerplate removal

File: crates/holochain/tests/inline_zome_sketch.rs [hunk 2]
```diff
@@ -34,145 +48,109 @@ fn simple_crud_zome() -> InlineZome {
...
+    let conductor: TestConductorHandle = Conductor::builder().test(&envs).await?.into();
+
+    // Install DNA and install and activate apps in conductor
+    let ids = conductor
+        .setup_app_for_all_agents_with_no_membrane_proof(
+            "app",
+            &[alice.clone(), bobbo.clone()],
+            &[dna_file],
+        )
+        .await;
```
Comment (maackle) [inline crates/holochain/tests/inline_zome_sketch.rs:70]: Here's where we install DNAs and apps for multiple agents and activate and set them up in one call.

File: crates/holochain/tests/inline_zome_sketch.rs [hunk 3]
```diff
@@ -34,145 +48,109 @@ fn simple_crud_zome() -> InlineZome {
...
 
-    let conductor = Conductor::builder().test(&envs).await?;
+    let ((alice,), (bobbo,)) = destructure_test_cells!(ids);
```
Comment (maackle) [inline crates/holochain/tests/inline_zome_sketch.rs:72]: And here's where we use some amazingly handle itertools magic to destructure the whole thing as tuples of arbitrary arity (up to 4)

File: crates/holochain/tests/inline_zome_sketch.rs [hunk 4]
```diff
@@ -34,145 +48,109 @@ fn simple_crud_zome() -> InlineZome {
...
-            .try_into()?
-    };
-
-    // Wait long enough for Bob to receive gossip
+    Ok(())
+}
+
+#[tokio::test(threaded_scheduler)]
+#[cfg(feature = "test_utils")]
+async fn inline_zome_3_agents_2_dnas() -> anyhow::Result<()> {
```
Comment (maackle) [inline crates/holochain/tests/inline_zome_sketch.rs:98]: The diff in this file would have been much more dramatic (and helpful) if I hadn't added a second test. Note the huge boilerplate removal of the first test, though.

**Review by freesig — APPROVED** [review]

**Pull request merged**